lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
> Can you add test cases with a hex property and a floating point property with a capital  E ?

I'll split up that test with the numeric literals as a separate test case as well, as it is getting a bit too long to read.